monai/handlers/checkpoint_saver.py

```diff
@@ -0,0 +1,194 @@
+        self.save_final = save_final
+        self.save_key_metric = save_key_metric
+        self.key_metric_name = key_metric_name
+        self.key_metric_n_saved = key_metric_n_saved
+        self.epoch_save_interval = epoch_save_interval
+        self.epoch_n_saved = epoch_n_saved
+        self.iteration_save_interval = iteration_save_interval
+        self.iteration_n_saved = iteration_n_saved
+
+        if self.save_final is True:
+
+            def _final_func(engine):
+                return engine.state.iteration
+
+            self.final_checkpoint = ModelCheckpoint(
+                self.save_dir,
+                file_prefix,
+                score_function=_final_func,
+                score_name="final_iteration",
+                require_empty=False,
+            )
+        if self.save_key_metric is True:
+
+            def _score_func(engine):
+                if isinstance(self.key_metric_name, str):
+                    metric_name = self.key_metric_name
+                elif hasattr(engine.state, "key_metric_name") and isinstance(engine.state.key_metric_name, str):
+                    metric_name = engine.state.key_metric_name
+                else:
+                    raise ValueError("must provde key_metric_name to save best validation model.")
+                return round(engine.state.metrics[metric_name], 4)
+
+            self.key_metric_checkpoint = ModelCheckpoint(
+                self.save_dir,
+                file_prefix,
+                score_function=_score_func,
+                score_name="key_metric",
+                n_saved=self.key_metric_n_saved,
+                require_empty=False,
+            )
+        if self.epoch_save_interval > 0:
+
+            def _epoch_func(engine):
+                return engine.state.epoch
+
+            self.epoch_checkpoint = ModelCheckpoint(
+                self.save_dir,
+                file_prefix,
+                score_function=_epoch_func,
+                score_name="epoch",
+                n_saved=self.epoch_n_saved,
+                require_empty=False,
+            )
+        if self.iteration_save_interval > 0:
+
+            def _iteration_func(engine):
+                return engine.state.iteration
+
+            self.iteration_checkpoint = ModelCheckpoint(
+                self.save_dir,
+                file_prefix,
+                score_function=_iteration_func,
+                score_name="iteration",
+                n_saved=self.iteration_n_saved,
+                require_empty=False,
+            )
+
+    def attach(self, engine):
+        if self.logger is None:
+            self.logger = engine.logger
+        if not engine.has_event_handler(self.completed, Events.COMPLETED):
+            engine.add_event_handler(Events.COMPLETED, self.completed)
+        if not engine.has_event_handler(self.epoch_completed, Events.EPOCH_COMPLETED):
+            engine.add_event_handler(Events.EPOCH_COMPLETED, self.epoch_completed)
```

Comment:
should we use the interface `Events.ITERATION_COMPLETED(every=n_iter)`? the LGTM tool will complain though...

Reply:
Do you mean change to:
engine.add_event_handler(Events.EPOCH_COMPLETED(every=1), self.epoch_completed)

I think `every=1` is default value, right?
Thanks.

Reply:
maybe i'm wrong, but i was thinking `engine.add_event_handler(Events.EPOCH_COMPLETED(every=self.epoch_save_interval), self.epoch_completed)`

Reply:
OK, I see. If LGTM prefers that, I will change in next patch.
Thanks.

Reply:
LGTM actually prefers the current way, `Events.EPOCH_COMPLETED(every=self.epoch_save_interval)` raises a warning about `EPOCH_COMPLETED` not callable...

Reply:
So..should we consider the LGTM warning and keep the current way?

Reply:
yes, i think `Events.EPOCH_COMPLETED(every=self.epoch_save_interval)` improves code readability

Reply:
updated in the latest patch.
Thanks.